include/gsl/multi_span

```diff
@@ -79,7 +79,7 @@ namespace details
     template <typename SizeType>
     struct SizeTypeTraits
     {
-        static const SizeType max_value = std::numeric_limits<SizeType>::max();
+        static const SizeType max_value = (std::numeric_limits<SizeType>::max)();
```

Comment:
I know about `#define NOMINMAX` trick. And I completely understand that it is windows.h fault. However some libraries for windows (for example, GDI+) rely on presence of such macro definition.
What do you think about removing SizeTypeTraits from GSL? It is not anymore.